src/mam4xx/ndrop.hpp

```diff
@@ -130,6 +245,1506 @@ void maxsat(
+                                    (latvap / (rh2o * tair) - one));
+  const Real beta = two * pi * rhoh2o * gthermfac * gamma; // [m2/s]
+  const Real wnuc = w_in;
+  const Real alw = alpha * wnuc;                  // [/s]
+  const Real etafactor1 = alw * haero::sqrt(alw); // [/ s^(3/2)]
+  // [unitless]
+  const Real zeta = twothird * haero::sqrt(alw) * aten / haero::sqrt(gthermfac);
+
+  Real amcube[nmode] = {}; // cube of dry mode radius [m3]
+
+  Real etafactor2[nmode] = {};
+  Real lnsm[nmode] = {};
+
+  // critical supersaturation for number mode radius [fraction]
+  Real ssat_crit_imode[nmode] = {};
+
+  Real eta[nmode] = {};
+  // Here compute ssat_crit_imode, eta for all modes for maxsat calculation
+  for (int imode = 0; imode < nmode; ++imode) {
+    // BAD CONSTANT
+
+    if (volume[imode] > small && na[imode] > small) {
+      // number mode radius [m]
+      // only if variable size dist
+      amcube[imode] =
+          three_fourths * volume[imode] / (pi * exp45logsig[imode] * na[imode]);
+      // Growth coefficient Abdul-Razzak & Ghan 1998 eqn 16
+      // should depend on mean radius of mode to account for gas kinetic
+      // effects see Fountoukis and Nenes, JGR2005 and Meskhidze et al.,
+      // JGR2006 for appropriate size to use for effective diffusivity.
+      etafactor2[imode] = one / (na[imode] * beta * haero::sqrt(gthermfac));
+      // BAD CONSTANT
+      if (hygro[imode] > 1.0e-10) {
+        ssat_crit_imode[imode] =
+            two * aten *
+            // only if variable size dist
+            haero::sqrt(aten / (27.0 * hygro[imode] * amcube[imode]));
+      } else {
+        // BAD CONSTANT
+        ssat_crit_imode[imode] = 100.0;
+      } // hygro
+    } else {
+      ssat_crit_imode[imode] = one;
+      // this should make eta big if na is very small.
+      etafactor2[imode] = etafactor2max;
+    } // volume
+    // only if variable size dist
+    lnsm[imode] = haero::log(ssat_crit_imode[imode]);
+    eta[imode] = etafactor1 * etafactor2[imode];
+  } // end imode
+
+  Real supersat = zero;
+
+  maxsat(zeta, eta, nmode, ssat_crit_imode, supersat);
+  // ([fraction]))
+  const Real lnsupersat = haero::log(supersat);
+
+  // Use maximum supersaturation to calculate aerosol activation output
+  for (int imode = 0; imode < nmode; ++imode) {
+    // [unitless]
+    const Real arg_erf_n =
+        twothird * (lnsm[imode] - lnsupersat) / (sq2 * alogsig[imode]);
+
+    fn[imode] = half * (one - haero::erf(arg_erf_n)); // activated number
+
+    const Real arg_erf_m = arg_erf_n - 1.5 * sq2 * alogsig[imode];
+    fm[imode] = half * (one - haero::erf(arg_erf_m)); // activated mass
+    fluxn[imode] = fn[imode] * w_in; // activated aerosol number flux
+    fluxm[imode] = fm[imode] * w_in; // activated aerosol mass flux
+  }
+  // NOTE: what is this??
+  // is vertical velocity equal to flux of activated aerosol fraction assuming
+  // 100% activation [m/s]?
```

Comment:
This seems to be true, based on the comment describing the parameter at the beginning of the function.